godot-core/src/builtin/rect2.rs

```diff
@@ -44,6 +44,180 @@ impl Rect2 {
         }
     }
 
+    /// Returns a rectangle with equivalent position and area, modified so that the top-left corner is the origin and `width` and `height` are positive.
+    ///
+    /// _Godot equivalent: `Rect2.abs()`_
+    #[inline]
+    pub fn abs(&self) -> Self {
+        Self {
+            position: self.position + Vector2::new(self.size.x.min(0.0), self.size.y.min(0.0)),
+            size: self.size.abs(),
+        }
+    }
+
+    /// Returns true if this rectangle (inclusively) encloses `b`. This is true when `self` covers all the area of `b`, and possibly (but not necessarily) more.
+    ///
+    /// _Godot equivalent: `Rect2.encloses(Rect2 b)`_
+    #[inline]
+    pub fn encloses(&self, b: Rect2) -> bool {
+        b.position.x >= self.position.x
+            && b.position.y >= self.position.y
+            && b.position.x + b.size.x <= self.position.x + self.size.x
+            && b.position.y + b.size.y <= self.position.y + self.size.y
```

Comment:
could use `b.end().x <= self.end().x` and same for y

Reply:
Sure, that will look more concise.